components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs

```diff
@@ -570,8 +570,9 @@ internal sealed class SubscriptionWrapper : ISubscription {
                 var revisedKeepAliveCount = _subscription.Configuration.KeepAliveCount
                     .GetValueOrDefault(session.DefaultSubscription.KeepAliveCount);
                 _subscription.MonitoredItems?.ForEach(m => {
-                    if (m.HeartbeatInterval != null && m.HeartbeatInterval != TimeSpan.Zero) {
-                        var itemKeepAliveCount = (uint)m.HeartbeatInterval.Value.TotalMilliseconds /
+                    var dataModel = m as DataMonitoredItemModel;
+                    if (dataModel != null && dataModel.HeartbeatInterval != null && dataModel.HeartbeatInterval != TimeSpan.Zero) {
```

Comment:
Just a detail, but you can combine the cast and check in the if statement:

```cs
if (m is DataMonitoredItemModel dataModel && ...
```

Reply:
Fixed